### PR TITLE
Add struct breaking change

### LINKED ISSUE #9148: Breaking changes in .NET Core 2.1 structs
There are breaking changes in some .NET Core 2.1 structs - these structs now have private fields and cannot be instantiated without `new`. There should be probabaly some official "breaking changes" section in this document.
https://github.com/dotnet/corefx/issues/6185

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 3bc74517-2839-4b16-8539-d31aba49bcfd
* Version Independent ID: 5511dd8b-d5d6-a3fb-172a-f35ff530cfc8
* Content: [What's new in .NET Core 2.1](https://docs.microsoft.com/en-us/dotnet/core/whats-new/dotnet-core-2-1)
* Content Source: [docs/core/whats-new/dotnet-core-2-1.md](https://github.com/dotnet/docs/blob/master/docs/core/whats-new/dotnet-core-2-1.md)
* Product: **dotnet-core**
* GitHub Login: @rpetrusha
* Microsoft Alias: **ronpet**

### PR DESCRIPTION
**Pull request by gewarren**
Fixes #9148

Preview link: https://review.docs.microsoft.com/en-us/dotnet/core/compatibility/2.0-2.1?branch=pr-en-us-16312

### CODE REVIEW

File: docs/csharp/structs.md [hunk 1]
```diff
@@ -9,43 +9,46 @@ ms.assetid: a7094b8c-7229-4b6f-82fc-824d0ea0ec40
...
+- A struct cannot declare a parameterless constructor (a constructor without parameters) or a finalizer.
+
+- Structs are copied on assignment. When a struct is assigned to a new variable, all the data is copied, and any modification to the new copy does not change the data for the original copy. This is important to remember when working with collections of value types such as Dictionary<string, myStruct>.
+
+- Structs are value types and classes are reference types.
+
+- Unlike classes, structs can be instantiated without using a `new` operator.
+
+   > [!NOTE]
+   > In .NET Core 2.1 and later, structs must be instantiated by using a `new` operator. For more information, see [Breaking changes for migration from version 2.0 to 2.1](../core/compatibility/2.0-2.1.md#corefx).
```
Comment (gewarren) [inline docs/csharp/structs.md]: Is there any change for .NET Framework that we need to note?

File: includes/core-changes/corefx/3.0/instantiate-struct.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+}
+```
+
+#### Category
+
+CoreFx
+
+#### Affected APIs
+
+- <xref:System.ConsoleKeyInfo>
```
Comment (gewarren) [inline includes/core-changes/corefx/3.0/instantiate-struct.md]: This is not a complete list, but https://github.com/dotnet/corefx/issues/6185 contains breadcrumbs for all the APIs that were changed. Do we need to add them all here and is there an easy way to obtain them?

File: includes/core-changes/corefx/3.0/instantiate-struct.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+- <xref:System.Memory%601?displayProperty=nameWithType>
+- <xref:System.Collections.Immutable.ImmutableArray%601>
+
+<!--
+
+### Affected APIs
+
+- `T:System.ConsoleKeyInfo`
+- `T:System.Memory{T}`
```
Comment (gewarren) [inline includes/core-changes/corefx/3.0/instantiate-struct.md]: Is this the right syntax for a generic type?

**Review by tdykstra — APPROVED** [review]
LGTM

File: includes/core-changes/corefx/2.1/instantiate-struct.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
+### Private fields added to built-in struct types
+
+Private fields were added to built-in struct types in [reference assemblies](~/standard/assembly/reference-assemblies.md). As a result, in C#, struct types must always be instantiated with the `new` operator.
+
+#### Change description
+
+In .NET Core 2.0 and previous versions, some built-in struct types, for example, <xref:System.ConsoleKeyInfo>, could be instantiated without using the `new` operator in C#. This was because the [reference assemblies](~/standard/assembly/reference-assemblies.md) used by the C# compiler didn't contain the private fields for the structs. All private fields for .NET struct types are added to the reference assemblies starting in .NET Core 2.1.
+
+For example, the following C# code compiles in .NET Core 2.0, but not in .NET Core 2.1:
```
Comment (BillWagner) [inline includes/core-changes/corefx/2.1/instantiate-struct.md]: Is it worth adding a second example with arrays? Something like the following:

```csharp
ConsoleKeyInfo[] keys = new ConsoleKeyInfo[5];
// The following is now needed:
for(int i = 0; i < keys.Length; i++)
    keys[i] = new ConsoleKeyInfo();
```

File: docs/csharp/structs.md [hunk 5]
```diff
@@ -9,43 +9,46 @@ ms.assetid: a7094b8c-7229-4b6f-82fc-824d0ea0ec40
...
-  
-- Structs can declare constructors that have parameters.  
-  
-- A struct cannot inherit from another struct or class, and it cannot be the base of a class. All structs inherit directly from <xref:System.ValueType>, which inherits from <xref:System.Object>.  
-  
-- A struct can implement interfaces.
+
+But you declare and assign values to them as if they were simple non-aggregate types:
+
+[!code-csharp[Assign Values](../../samples/snippets/csharp/concepts/structs/assign-value.cs)]
```
Comment (jaredpar) [inline docs/csharp/structs.md]: Is this markdown form specific to the docs website? It looks quite promising. Wondering if I can use this anywhere ...

File: docs/csharp/structs.md (hunk 1 above)
Comment (jaredpar) [reply docs/csharp/structs.md]: > In .NET Core 2.1 and later, structs must be instantiated by using a `new` operator

This is not quite correct. A struct must be instantiated by either `new`, `default` or by initializing all of the public fields. The latter is the rare case though, bordering on advanced level.

File: docs/csharp/structs.md (hunk 5 above)
Comment (gewarren) [reply docs/csharp/structs.md]: I believe the code snippet reference is specific to docs.

Comment (jaredpar) [reply docs/csharp/structs.md]: That's unfortunate cause it seems really neat.

**Pull request merged**